src/script/bitcoinconsensus.cpp

```diff
@@ -99,7 +99,13 @@ static int verify_script(const uint8_t *scriptPubKey,
         // Regardless of the verification result, the tx did not error.
         set_error(err, bitcoinconsensus_ERR_OK);
 
-        PrecomputedTransactionData txdata(tx);
+        // Note: The new BIP341 sighash requires knowing all CTxOut of the tx's
+        // inputs. We don't have access to this here yet, so verification of
+        // transactions that rely on the outputs will fail.
+        // Eventually we should add a
+        // bitcoinconsensus_verify_script_with_spent_outputs function to address
+        // this.
+        PrecomputedTransactionData txdata(tx, {});
```

Comment:
I'd make sure you had access to the coinsview or whatever you need here first. That way there can't be a mistake later.